server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -782,20 +783,28 @@ private void findStorageGroup(MNode node, String path, HashSet<String> res) {
    *
    * @return a list contains all distinct devices
    */
-  Set<String> getAllDevices() {
-    return new HashSet<>(getNodesList(3));
+  Set<String> getAllDevices() throws SQLException {
+    return new HashSet<>(getNodesList("root",2));
```

Comment:
`device` is not necessarily at level 2.
`device` is at the secod to last level. 
For example, `CREATE TIMESERIES root.demo.m2.m2.s0 WITH DATATYPE=INT32,ENCODING=RLE` and the device should be `root.demo.m2.m2`.

Reply:
@LeiRui Hi, actually I have no idea now about how to implement this interface as it was not developed by me before. What I done is just making sure that the program does not report the error.

However, I would like to discuss with you guys about how to make changes of this function,  could you please give me some advices? I think this should create another PR for this.

Reply:
Hi, terribly sorry for the late reply, and I’m responsible for this issue. Do you think it is necessary to add a new param `String schemaPattern` for method `Set<String> getAllDevices()`  as well?

Reply:
Or maybe I will fix this bug after your PR is merged so that it won't result in too many conflicts since your PR is currently not reporting any error.